Tests/ModelHelpers/AttributesTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @copyright  Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Entity\ModelHelpers\Tests;
+
+use Joomla\Entity\Exceptions\AttributeNotFoundException;
+use Joomla\Entity\Helpers\Collection;
+use Joomla\Entity\Relations\Relation;
+use Joomla\Entity\Tests\Models\User;
+use Joomla\Entity\Tests\SqliteCase;
+use Joomla\Entity\Model;
+
+/**
+ * @since  1.0
+ */
+class AttributesTest extends SqliteCase
+{
+
+	/**
+	 * This method is called before the first test of this test class is run.
+	 * @return void
+	 */
+	public static function setUpBeforeClass()
+	{
+		static::$dataSets = array(
+			'banners'       => __DIR__ . '/Stubs/banners.csv',
+			'messages'      => __DIR__ . '/Stubs/messages.csv',
+			'users'         => __DIR__ . '/Stubs/users.csv',
+			'user_profiles' => __DIR__ . '/Stubs/user_profiles.csv'
+		);
+
+		parent::setUpBeforeClass();
+	}
+
+	/**
+	 * @covers \Joomla\Entity\ModelHelpers\Attributes::getAttribute()
+	 * @return void
+	 */
+	public function testGetAttribute()
+	{
+		$userModel = new User(self::$driver);
+
+		$user = $userModel->find(42);
+
+		$simpleAttribute = $user->getAttribute('id');
+
+		$mutatorAttribute = $user->getAttribute('newAccount');
+
+		$dateAttribute = $user->getAttribute('registerDate');
+
+		$castAttribute = $user->getAttribute('params');
+
+		$relationAttribute = $user->getAttribute('sentMessages');
+
+		$this->assertEquals(
+			42,
+			$simpleAttribute
+		);
+
+		$this->assertTrue(
+			$mutatorAttribute
+		);
+
+		$this->assertEquals(
+			'2010-02-13 00:34:42',
+			$dateAttribute
+		);
+
+		$this->assertEquals(
+			['test' => 'Object'],
+			$castAttribute
+		);
+
+		$this->assertInstanceOf(
+			Collection::class,
+			$relationAttribute
+		);
+
+		$this->assertInstanceOf(
+			Model::class,
+			$relationAttribute->first()
+		);
+
+		$this->assertEquals(
+			1,
+			$relationAttribute->first()->message_id
+		);
+
+		$this->expectException(AttributeNotFoundException::class);
```

Comment:
Please put the `AttributeNotFoundException` tests into a separate function test - this way we ensure that no valid attribute is throwing a `AttributeNotFoundException`